src/commands/midi.ts

```diff
@@ -36,15 +37,95 @@ function playMidi(contextString: string, editor: TextEditor) {
 	else
 		chordType = lineContext(contextString);
 
+	//turns chordtype name into a note for flag sounds
```

Comment:
These lines can be removed in favor of calling the adjusted playFlagMidi function within the if/else block.

If audioFlagNote is not undefined, and thus returns true, you can get the note and instrument values from the FlagSound class and pass them to the playFlagMidi function.

In the case that the audioFlagNote is unedefined, you can pass the result of `lineContext(contextString)` to the playFlagMidi function.